### PR TITLE
net: packet clone - add copying of missing flags inside clone_pkt_attributes() function.

### PR DESCRIPTION
**Pull request by MarGasiorek**
Add cloning of missing flags (attributes).
Update UT.
Small code improvement.

### CODE REVIEW

File: include/zephyr/net/net_pkt.h [hunk 1]
```diff
@@ -509,7 +509,7 @@ static inline void net_pkt_set_eof(struct net_pkt *pkt, bool eof)
 #if defined(CONFIG_NET_ROUTE)
```
Comment (jukkar) [inline include/zephyr/net/net_pkt.h:509]: The commit subject `net: Code improvement` is too generic. We could change that to
`net: Follow coding style when converting int to bool`

File: tests/net/net_pkt/prj.conf [hunk 2]
```diff
@@ -4,11 +4,13 @@ CONFIG_ZTEST_NEW_API=y
...
+CONFIG_NET_IPV6=y
 CONFIG_NET_ARP=y
 CONFIG_NET_UDP=y
 CONFIG_NET_BUF_LOG=y
 CONFIG_NET_LOG=y
 CONFIG_NET_BUF_POOL_USAGE=y
 CONFIG_ENTROPY_GENERATOR=y
 CONFIG_TEST_RANDOM_GENERATOR=y
+CONFIG_NET_IPV6_NBR_CACHE=y
+CONFIG_NET_ETHERNET_BRIDGE=y
```
Comment (jukkar) [inline tests/net/net_pkt/prj.conf:16]: Why do we need bridging enabled?

File: include/zephyr/net/net_pkt.h (hunk 1 above)
Comment (MarGasiorek) [reply include/zephyr/net/net_pkt.h:509]: Sorry for that, I changed comment as you suggest.

File: tests/net/net_pkt/prj.conf (hunk 2 above)
Comment (MarGasiorek) [reply tests/net/net_pkt/prj.conf:16]: I enabled it in order to test l2_bridged flag:
https://github.com/zephyrproject-rtos/zephyr/blob/d42d14e392e70a846abe5baaceb849ef21a50584/include/zephyr/net/net_pkt.h#L394

Comment (jukkar) [reply tests/net/net_pkt/prj.conf:16]: Ok, was just wondering about that because it was not very clear.

**Review by jukkar — DISMISSED** [review]

**Review by ssharks — DISMISSED** [review]

File: subsys/net/ip/net_pkt.c [hunk 3]
```diff
@@ -1809,6 +1809,13 @@ static void clone_pkt_attributes(struct net_pkt *pkt, struct net_pkt *clone_pkt)
 	net_pkt_set_priority(clone_pkt, net_pkt_priority(pkt));
 	net_pkt_set_orig_iface(clone_pkt, net_pkt_orig_iface(pkt));
 	net_pkt_set_captured(clone_pkt, net_pkt_is_captured(pkt));
+	net_pkt_set_sent(clone_pkt, net_pkt_sent(pkt));
+	net_pkt_set_queued(clone_pkt, net_pkt_queued(pkt));
```
Comment (rlubos) [inline subsys/net/ip/net_pkt.c:1813]: Is this correct to clone those flags? If the original packet was marked as sent/queued, it does not seem to be true for the clone.

Anyway, both these flags seem no longer in use (they aren't read anywhere within the codebase). The setters are called in a couple of places, but according to the comments/git messages they seemed to have been used by the old TCP implementation (some comments even refer to obsolete TCP code).

Maybe if we're doing cleanup in the flags, we should consider removing them instead?

Comment (jukkar) [reply subsys/net/ip/net_pkt.c:1813]: > Is this correct to clone those flags? If the original packet was marked as sent/queued, it does not seem to be true for the clone.

I was thinking that too, but it is difficult to know what flags would not need to be cloned. The cloning implies also that we get identical copy of the original net_pkt so with in mind I considered it ok if we just copy all the flags.

> 
> Anyway, both these flags seem no longer in use (they aren't read anywhere within the codebase). The setters are called in a couple of places, but according to the comments/git messages they seemed to have been used by the old TCP implementation (some comments even refer to obsolete TCP code).

The sent flag is shared with eof, so while we could remove the function, the actual bit field cannot be removed (but could be renamed).

> 
> Maybe if we're doing cleanup in the flags, we should consider removing them instead?

Yes, that is good idea.

File: subsys/net/ip/net_pkt.c [hunk 4]
```diff
@@ -1809,6 +1809,13 @@ static void clone_pkt_attributes(struct net_pkt *pkt, struct net_pkt *clone_pkt)
...
+	net_pkt_set_ptp(clone_pkt, net_pkt_is_ptp(pkt));
+
+#if defined(CONFIG_NET_ROUTE)
```
[the remark this answers is not in the document]
Comment (jukkar) [reply subsys/net/ip/net_pkt.c:1816]: Good point, +1 to that.

File: subsys/net/ip/net_pkt.c (hunk 3 above)
Comment (MarGasiorek) [reply subsys/net/ip/net_pkt.c:1813]: > 

> Maybe if we're doing cleanup in the flags, we should consider removing them instead?

> Yes, that is good idea.

If this flag is really unused then maybe should be renamed to `unused` for future use.

File: subsys/net/ip/net_pkt.c (hunk 4 above)
[the remark this answers is not in the document]
Comment (MarGasiorek) [reply subsys/net/ip/net_pkt.c]: The `#if defined(CONFIG_NET_ROUTE)` has been removed.

File: subsys/net/ip/net_pkt.c (hunk 3 above)
Comment (MarGasiorek) [reply subsys/net/ip/net_pkt.c]: I removed all deprecated and unused flags together with setters/getters.

**Review by jukkar — COMMENTED** [review]
I would prefer that the commit "net: ip: Add cloning of missing attributes" is split to two. First one that does the removal of things, and the second one that does the cloning.

**Comment by MarGasiorek** [comment]
> I would prefer that the commit "net: ip: Add cloning of missing attributes" is split to two. First one that does the removal of things, and the second one that does the cloning.

Done

**Review by jukkar — APPROVED** [review]

**Review by rlubos — APPROVED** [review]
Thanks

**Pull request merged**